IPython/terminal/shortcuts/__init__.py

```diff
@@ -451,34 +450,38 @@ def newline_or_execute(event):
         if d.line_count == 1:
             check_text = d.text
         else:
-            check_text = d.text[:d.cursor_position]
+            check_text = d.text[: d.cursor_position]
         status, indent = shell.check_complete(check_text)
-       
+
         # if all we have after the cursor is whitespace: reformat current text
         # before cursor
-        after_cursor = d.text[d.cursor_position:]
+        after_cursor = d.text[d.cursor_position :]
         reformatted = False
         if not after_cursor.strip():
             reformat_text_before_cursor(b, d, shell)
             reformatted = True
-        if not (d.on_last_line or
-                d.cursor_position_row >= d.line_count - d.empty_line_count_at_the_end()
-                ):
+        if not (
+            d.on_last_line
+            or d.cursor_position_row >= d.line_count - d.empty_line_count_at_the_end()
+        ):
             if shell.autoindent:
-                b.insert_text('\n' + indent)
+                b.insert_text("\n" + indent)
             else:
-                b.insert_text('\n')
+                b.insert_text("\n")
             return
 
-        if (status != 'incomplete') and b.accept_handler:
+        if (status != "incomplete") and b.accept_handler:
             if not reformatted:
                 reformat_text_before_cursor(b, d, shell)
             b.validate_and_handle()
         else:
             if shell.autoindent:
-                b.insert_text('\n' + indent)
+                b.insert_text("\n" + indent)
             else:
-                b.insert_text('\n')
+                b.insert_text("\n")
+
+    newline_or_execute.__qualname__ = "newline_or_execute"
```

Comment:
So that gives this a better name and remove the `<local>`, but is there any other reasons ? Same questions for the other `__name__`, like `is_focused...`, or `preceding_text(...)`.

Reply:
It is just for a nicer name:
- in the documentation as it is used in the table,
- in future so that we can use it to identify the keybinding and change it.